File: bench/dotnet/bintree.cs
namespace Bench;
sealed class Node { public Node L, R; }
static class Bintree
{
    static Node Make(int d) => d == 0 ? null : new Node { L = Make(d - 1), R = Make(d - 1) };
    static long Check(Node node) => node == null ? 1 : 1 + Check(node.L) + Check(node.R);
    public static void Run(int n)
    {
        const int DEPTH = 12;
        long total = 0;
        for (int i = 0; i < n; i++) total += Check(Make(DEPTH));
        Console.WriteLine(total);
    }
}

File: bench/clojure/bintree.clj
(def depth 12)
(defn make [^long d] (if (= d 0) nil [(make (- d 1)) (make (- d 1))]))
(defn check ^long [node] (if (nil? node) 1 (+ 1 (check (nth node 0)) (check (nth node 1)))))
(let [n (long (Long/parseLong (or (System/getenv "BENCH_N") "200")))]
  (println (loop [i 0 acc 0]
             (if (< i n) (recur (inc i) (+ acc (check (make depth)))) acc))))
